### DISCUSSION TITLE
inplace vector initialization

### DISCUSSION THREAD

**Discussion by iperov** [ispc/ispc#2914] [Ideas]
```c
typedef float<2> vec2;

vec2 p0 = {0, 0}; //fine

vec2 pt = vec2(0, 0); //error
vec2 pt1 = pt + vec2(1,1); //error 
```
could you add such a feature?

**Comment by dbabokin** [comment]
This C++-style initialization would be very nice feature to have and I tried to add in the past, but unfortunately it's impossible to implement (or at least I haven't found a viable solution after spending quite some time) with current parser (i.e. bison), as this grammar rules introduce ambiguity to the parser.

**Reply by dbabokin** [reply]
More details are here: https://github.com/ispc/ispc/issues/1279

**Reply by iperov** [reply]
nothing is impossibru! just do it

**Comment by iperov** [comment]
with this mini glsl lib 
https://github.com/re-esper/CPUShadertoy/blob/master/glsl.ispc

```
vec2 f = vec2(2,2);
```
works fine

**Reply by dbabokin** [reply]
Smart!

**Reply by iperov** [reply]
@dbabokin what if to make it part of the standard library?

**Reply by dbabokin** [reply]
For standard library it seems like half-baked solution. It immediately causes questions - why it works for some stuff and not for others. We better find a solution for that problem by fixing parser (switching to ANTLR, for example).
